irsa.tf

```diff
@@ -0,0 +1,18 @@
+# Enable IAM Roles for EKS Service-Accounts (IRSA).
+
+# Obtaining the Root CA Thumbprint for an OpenID Connect Identity Provider.
+# Requires: openssl as a dependecy.
+# https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html
+# https://github.com/terraform-providers/terraform-provider-aws/issues/10104
+
+data "external" "fetch_thumbprint" {
+  count   = var.enable_irsa ? 1 : 0
+  program = ["sh", "-c", "s_client -connect oidc.eks.${aws_region}.amazonaws.com:443 2>&- -showcerts | awk '/-----BEGIN/{f=\"/tmp/cert.\"(n++)} f{print>f} /-----END/{f=\"\"}' | openssl x509 -noout -fingerprint -in /tmp/cert.3 | cut -d'=' -f2 | tr -d ':' | awk '{print tolower($1)}' | jq -rR 'split(\" \")|{fingerprint:.[0]}'"]
```

Comment:
I don't think this is workable. For example many people don't have `jq` installed. Also won't work on Windows

Reply:
I have it already working for 2 clusters and tested it many times. But we use only Linux that's why I did not consider generalizing it to different OS. 

If the `jq` is the only problem, I think I can refactor this to output JSON result without the use of `jq`. Maybe we could mention on the README that this feature requires Linux OS to run on until the issue is resolved in Terraform or AWS side.

Reply:
So, It turned out that this Root CA certificate is valid until `Jun 28 17:39:16 2034 GMT`
https://crt.sh/?q=9E99A48A9960B14926BB7F3B02E22DA2B0AB7280
and is valid across all regions so passing it as a default value without dynamically calculating the certificate fingerprint would have a very low security impact. This way there wont be any external dependencies.